doc/article/en-US/templating-html-behaviors-introduction.md

```diff
@@ -297,7 +297,232 @@ The `@bindable` decorator isn't used when doing single value binding with a Cust
+  * `String` to `'string'`
+  * `Boolean` to `'boolean'`
+  * `Date` to `'date'`
+
+You can extend / modify this via export `coerceFunctionMap` of `aurelia-binding`:
+
+<code-listing heading="extend-coerce">
+  <source-code lang="TypeScript">
+    // import from 'aurelia-binding' if you are writing a plugin
+    // import from 'aurelia-framework' if you are writing an application
+    import {coerceFunctions, coerceFunctionMap} from 'aurelia-framework';
+
+    // Add `point` coerce function
+    coerceFunctions.point = function(incomingValue) {
+      return incomingValue.split(' ').map(parseFloat);
+    };
+
+    // Register a class to be used with coerce
+    coerceFunctionMap.set(Point, 'point');
+
+    // then we can simply have
+    export class Line {
+      @bindable point1: Point
+      @bindable point2: Point
+    }
+  </source-code>
+</code-listing>
+
+In above example, `bindable` decorator auto picks up the type, set proper coerce function for `point1` and `point2`.
+
+JavaScript user can also leverage this pattern via `Reflect.metadata` decorator by decorating property with correct metadata for `propertyType` before decorating it with `bindable`, for example:
+
+<code-listing heading="extend-coerce">
+  <source-code lang="ES 2015">
+    // import from 'aurelia-binding' if you are writing a plugin
+    // import from 'aurelia-framework' if you are writing an application
+    import {coerceFunctions, coerceFunctionMap} from 'aurelia-framework';
+    import {metadata} from 'aurelia-metadata';
+
+    // Add `point` coerce function
+    coerceFunctions.point = function(incomingValue) {
+      return incomingValue.split(' ').map(parseFloat);
+    };
+
+    // Register a class to be used with coerce
+    coerceFunctionMap.set(Point, 'point');
+
+    // then we can simply have
+    export class Line {
+      @bindable
+      @Reflect.metadata(metadata.propertyType, Point)
+      point1
+      
+      @bindable
+      @Reflect.metadata(metadata.propertyType, Point)
+      point2
+    }
+  </source-code>
+</code-listing>
+
+In above example, `bindable` decorator auto picks up the type, set proper coerce function for `point1` and `point2`.
+
+#### Fluent syntax
+
+If you scroll back top a bit, you will notice we had fluent syntax decorator: `@bindable.number`. This, as described above, is a simplified and more expressive form of `coerce: 'number'`. Aurelia also provides a way to build your own fluent syntax `bindable` decorator. You can do this by:
+
+<code-listing heading="extend-coerce">
+  <source-code lang="ES 2015">
+    import {coerceFunctions, coerceFunctionMap, createTypedBindable} from 'aurelia-framework';
+    
+    // This is to enable `@bindable.point`
+    createTypedBindable('point');
+
+    // Register a class to be used with coerce
+    coerceFunctionMap.set(Point, 'point');
+
+    // Add `point` coerce function
+    coerceFunctions.point = function(incomingValue) {
+      return incomingValue.split(' ').map(parseFloat);
+    };
```

Comment:
Perhapse there should be a helper to simplify registering fluent coerce syntax? For example instead of 
+ `import {coerceFunctions, coerceFunctionMap, createTypedBindable} from 'aurelia-framework';`
+ `createTypedBindable(..)`
+ `coerceFunctionMap.set(...)`
+ `coerceFunctions.... = function...` 
it could be done for example like this:
```TypeScript
import {coerceHelper} from 'aurelia-framework';
coerceHelper.addFuluentBindableExtension('point', Point, (incomingValue) => {
    return incomingValue.split(' ').map(parseFloat);
});
```

Reply:
Right, I was feeling this is too much call too, for API designed, I think @jdanyow @EisenbergEffect will need to give call on this.

@atsu85 Thanks for helping, I ran out of energy half way through the docs 😱 , many typos and unneeded stuff